main.py

```diff
@@ -93,23 +99,25 @@ def create_transaction(self):
         #     self.stop()
         #     return
 
-    def check_transactions(self):
-        print(f'[Node {self.get_peer_id(self.my_peer)}] Checking transactions')
-        logging.info(f'[Node {self.get_peer_id(self.my_peer)}] Checking transactions')
 
+    def block_creation(self):
+        # WIP: get every known peer, not only the ones im connected to
+        peers = self.get_peers()
+        peers.append(self.my_peer)
+        selectedPeer = random.choice(peers)
+
+        if not selectedPeer.mid == self.my_peer.mid:
+            return
 
         for tx in self.pending_txs:
-            if self.balances[tx.sender] - tx.amount >= 0:
```

Comment:
Do you want to add this part to on_transaction?

Reply:
Not to on transaction but to when we receive a block
In todays class Bulat explained that when we receive a block we have to do a stateful and stateless validation of the block, so this validation of the tx's would be at this point